### PR TITLE
chore(setup-tests)

### PR DESCRIPTION
**Pull request by HugoDF**
First part of #37 

- add testing dependencies (jest, jest-dom, dom-testing-library, babel-preset-env)
- configure babel with preset-env for testing
- create util to load up `panel.html` body without the panel.js script include
- create util to generate a component object as expected from State
- add initialisation sanity test (ie. Alpine starts up correctly on an empty panel.html)
- add component-list sanity tests (injected components are visible, clicking on a component brings up the data pane with right values)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -22,6 +22,8 @@ jobs:
         node-version: ${{ matrix.node-version }}
     - name: install
       run: npm ci
+    - name: test
+      run: npm t
```
Comment (ryangjchandler) [inline .github/workflows/build.yml:26]: `npm t` instead of `npm test`, didn't know you could do this?

Comment (HugoDF) [reply .github/workflows/build.yml:26]: I'm tempted to rename the `test` folder to `tests` to keep it in line with Alpine 🤔

Comment (HugoDF) [reply .github/workflows/build.yml:26]: oh wait it's already `tests` 😂

**Comment by HugoDF** [comment]
@Te7a-Houdini I'm merging this since it doesn't change any functionality + it's only 1 of many PRs for adding tests, let me know if you have any feedback here or on the issue 😄

**Pull request merged**